src/components/substance/SubstanceForm.vue

```diff
@@ -147,21 +147,29 @@ export default {
       return this.$store.state.substance.form;
     },
     sourceOptions: function() {
-      return this.buildOptions(this.sourceList);
+      return this.buildOptions(this.sourceList, this.form.sourceID);
     },
     substanceTypeOptions: function() {
-      return this.buildOptions(this.substanceTypeList);
+      return this.buildOptions(
+        this.substanceTypeList,
+        this.form.substanceTypeID
+      );
     },
     qcLevelsOptions: function() {
-      return this.buildOptions(this.qcLevelList);
+      return this.buildOptions(this.qcLevelList, this.form.qcLevelID);
     }
   },
   methods: {
-    buildOptions: function(list) {
+    buildOptions: function(list, selected) {
       let item;
       let options = [];
       for (item of list)
-        options.push({ value: item.id, text: item.attributes.label });
+        if (!item.attributes.deprecated || item.id === selected)
```

Comment:
why should it matter that the item is deprecated? we don't want to filter out the item from the options list we just want to disable it, right? this is likely going to be changed with my substance saving ticket. I would think we still want to push all of the options but with some of them being disabled. Otherwise, I am not seeing any of the deprecated and won't the loading of a substance with a value for one of these fields that is deprecated cause a fail? I've moved the creation of options to a getter in each respective store, a similar thing might be able to be done with QST and what is needed in the agGrid tables. the form-select has a slot for the other elements we need for QST, like None and DefinedCompound.

Reply:
why should it matter that the item is deprecated? we don't want to filter out the item from the options list we just want to disable it, right?

I was trying to address this in the definition of done.